tests/stdlib/types/tpackedsets.nim

```diff
@@ -235,15 +235,15 @@ block legacyMainModuleTests:
 
       proc bug12366 =
         var
-          x = initPackedSet[A]()
+          a = initPackedSet[A]()
```

Comment:
No need to change it back, just curious why the rename?

Reply:
It's the symbol re-definition issue again (although in a much more subtle way).

`x` is also the name of a `typed` argument of the template this procedure is defined inside. Below, the template is invoked with non-top-level globals passed to `x`.

This leads to the `var substitutedGlobalSymbolGoesHere` to be treated as a procedure-level global. If those have a destructor (`legacyInit`, `instGenericInit`, etc. do have one), they're registered in the global destructor list, which in this case caused the globals to be destroyed twice (a double free that resulted in a crash).